go/src/koding/klient/machine/index/cache.go

```diff
@@ -10,12 +10,27 @@ import (
 	"path/filepath"
 	"strings"
 	"time"
+
+	"koding/klient/machine/index/filter"
 )
 
 // TempIndexDirPrefix defines the prefix of temporary index file created by
 // Cached structure.
 const TempIndexDirPrefix = "koding_index_"
 
+// DefaultFilter defines a default filter used to process index caching.
+var DefaultFilter filter.Filter = filter.MultiFilter{
+	filter.OsSkip(filter.DirectorySkip(".Trash"), "darwin"),                                                                   // OSX trash directory.
+	filter.OsSkip(filter.DirectorySkip(".Trashes"), "darwin"),                                                                 // OSX trash directory.
+	filter.NewWithError(filter.PathSuffixSkip(".git/index.lock"), "git repository on remote is in use"),                       // git index lock file.
```

Comment:
@cihangir `filter.NewWithError(` means that if these files are present on remote, the local user who tries to mount such repository will get an error. In your case:
- rebase on remote
- rebase fails
- index.lock stays
- user tries to mount rebase-failed remote
- first mount step is to create cache index(what this logic does)
- filter detects `.git/index.lock` and since it's "WithError", creating index returns error
- nothing on remote happen.
- user who tried to mount will receive following error:
```
git repository on remote is in use (path: .git/index.lock)
```
- mounts will fail